src/docs/sphinx/tutorials/step12/Tutorial.rst

```diff
@@ -0,0 +1,357 @@
+With ``useCartesianOuterBoundary="0"``, a Cartesian aligned outer boundary on the outer block is enforced.
+In this way, a structured three-dimensional mesh is created with 100 x 80 x 2 elements in the radial, tangential and z directions, respectively. All the elements are eight-node hexahedral elements (``C3D8``) and refinement is performed
+to conform with the wellbore geometry. This mesh is defined as a cell block with the name
+``cb1``.
+
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+    :language: xml
+    :start-after: <!-- SPHINX_WELLBORE_MESH -->
+    :end-before: <!-- SPHINX_WELLBORE_MESH_END -->
+
+
+Solver: setting up the solid mechanics solver
+-----------------------------------------------------------
+
+GEOSX is a multi-physics platform. Different combinations of
+physics solvers available in the code can be applied
+in different regions of the domain and be functional at different stages of the simulation.
+The ``Solvers`` tag in the XML file is used to list and parameterize these solvers.
+
+To specify a coupling between two different solvers, we define and characterize each single-physics solver separately.
+Then, we customize a *coupling solver* between these single-physics
+solvers as an additional solver.
+This approach allows for generality and flexibility in constructing multi-physics solvers.
+The order of specifying these solvers is not restricted in GEOSX.
+Note that end-users should give each single-physic solver a meaningful and distinct name, as GEOSX will recognize these single-physics solvers based on their customized names and create user-expected coupling.
+
+As demonstrated in this tutorial, to setup a poro-mechanical coupling, we need to effectively define three different solvers in the XML file:
+
+- the small strain Lagrangian mechanics solver, a solver of type ``SolidMechanics_LagrangianFEM`` called here ``mechanicsSolver`` (more information here: :ref:`SolidMechanicsLagrangianFEM`),
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+  :language: xml
+  :start-after: <!-- SPHINX_WELLBORE_MECHANICALSOLVER -->
+  :end-before: <!-- SPHINX_WELLBORE_MECHANICALSOLVER_END -->
+
+
+- the single-phase flow solver, a solver of type ``SinglePhaseFVM`` called here ``SinglePhaseFlowSolver`` (more information on these solvers at :ref:`SinglePhaseFlow`),
+
+.. literalinclude:: ../../../../../examples/plasticity/WellboreProblem_PoroDruckerPrager.xml
+  :language: xml
+  :start-after: <!-- SPHINX_WELLBORE_SINGLEPHASEFVM -->
+  :end-before: <!-- SPHINX_WELLBORE_SINGLEPHASEFVM_END -->
+
+
+- the coupling solver (``Poroelastic``) that will bind the two single-physics solvers above, which is named as ``PoroelasticitySolver`` (more information at :ref:`PoroelasticSolver`).
```

Comment:
Poromechanics may be a better name given we are solving a plastic problem.

Reply:
Right, same solver (SinglePhasePoromechanics) is used for solving both poroelastic and poroplastic problems, whose major difference is the material model. To avoid confusion, PoromechanicsSolver should be used in this tutorial.